indico/legacy/webinterface/pages/conferences.py

```diff
@@ -96,35 +91,62 @@ def render_event_footer(event, dark=False):
 
 
-class WPConferenceBase(base.WPDecorated):
+class WPConferenceBase(WPDecorated):
     @unify_event_args
     def __init__(self, rh, event_, **kwargs):
         assert event_ == kwargs.setdefault('event', event_)
+        self.event = event_
         WPDecorated.__init__(self, rh, **kwargs)
         self._navigationTarget = self._conf = event_.as_legacy
         self._tz = event_.display_tzinfo.zone
         start_dt_local = event_.start_dt_display.astimezone(event_.display_tzinfo)
         end_dt_local = event_.end_dt_display.astimezone(event_.display_tzinfo)
-        dates = " (%s)" % format_date(start_dt_local, format='long')
-        if start_dt_local.strftime("%d%B%Y") != end_dt_local.strftime("%d%B%Y"):
-            if start_dt_local.strftime("%B%Y") == end_dt_local.strftime("%B%Y"):
-                dates = " (%s-%s)" % (start_dt_local.strftime("%d"), format_date(end_dt_local, format='long'))
+        dates = u' ({})'.format(to_unicode(format_date(start_dt_local, format='long')))
+        if start_dt_local.date() != end_dt_local.date():
+            if start_dt_local.year == end_dt_local.year and start_dt_local.month == end_dt_local.month:
+                dates = u' ({}-{})'.format(start_dt_local.day, format_date(end_dt_local, format='long'))
             else:
-                dates = " (%s - %s)" % (format_date(start_dt_local, format='long'),
-                                        format_date(end_dt_local, format='long'))
-        self._setTitle("%s %s" % (strip_ml_tags(self._conf.as_event.title.encode('utf-8')), dates))
+                dates = u' ({} - {})'.format(to_unicode(format_date(start_dt_local, format='long')),
+                                             to_unicode(format_date(end_dt_local, format='long')))
+        self._setTitle(u'{} {}'.format(strip_tags(self.event.title), dates))
 
     def _getHeader(self):
         raise NotImplementedError  # must be overridden by meeting/lecture and conference WPs
 
     def getJSFiles(self):
-        return base.WPDecorated.getJSFiles(self) + self._asset_env['modules_event_display_js'].urls()
+        return WPDecorated.getJSFiles(self) + self._asset_env['modules_event_display_js'].urls()
+
+    def _getHeadContent(self):
+        site_name = core_settings.get('site_title')
+        meta = render_template('events/meta.html', event=self.event, site_name=site_name)
+        return WPDecorated._getHeadContent(self) + meta
 
 
 class WPConferenceDefaultDisplayBase(MathjaxMixin, WPConferenceBase):
     menu_entry_plugin = None
     menu_entry_name = None
 
+    @unify_event_args
+    def __init__(self, rh, event_, **kwargs):
+        assert event_ == kwargs.setdefault('event', event_)
+        self.event = event_
+        kwargs['conf_layout_params'] = self._get_layout_params()
+        WPConferenceBase.__init__(self, rh, event_, **kwargs)
+
+    def _get_layout_params(self):
+        bg_color = layout_settings.get(self.event, 'header_background_color').replace('#', '').lower()
```

Comment:
why are you placing `#` here? you are using it anyway a few lines below

Reply:
```
In [4]: Counter(x.value.startswith('#') for x in EventSetting.query.filter_by(name='header_background_color').filter(db.cast(EventSetting.value, db.Text) != '""
   ...: '))
Out[4]: Counter({False: 20, True: 2075})

In [5]: Counter(x.value.startswith('#') for x in EventSetting.query.filter_by(name='header_text_color').filter(db.cast(EventSetting.value, db.Text) != '""'))
Out[5]: Counter({False: 5, True: 1983})
```